### PR TITLE
[SPARK-23143][SS][PYTHON] Added python API for setting continuous trigger

### PR DESCRIPTION
**Pull request by tdas**
## What changes were proposed in this pull request?
Self-explanatory.

## How was this patch tested?
New python tests.

### CODE REVIEW

**Comment by tdas** [comment]
@zsxwing @jose-torres please take a look.

**Comment by tdas** [comment]
jenkins retest this please

File: python/pyspark/sql/tests.py [hunk 1]
```diff
@@ -1518,6 +1518,12 @@ def test_stream_trigger(self):
         except ValueError:
             pass
 
+        # Should not take multiple args
+        try:
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests.py]: @tdas, could we maybe test this by `assertRaises` if it checks raising an exception?

File: python/pyspark/sql/streaming.py [hunk 2]
```diff
@@ -802,23 +802,38 @@ def trigger(self, processingTime=None, once=None):
         >>> writer = sdf.writeStream.trigger(processingTime='5 seconds')
         >>> # trigger the query for just once batch of data
         >>> writer = sdf.writeStream.trigger(once=True)
+        >>> # trigger the query for execution every 5 seconds
```
Comment (jose-torres) [inline python/pyspark/sql/streaming.py]: nit: not quite the right comment

Comment (jose-torres) [reply python/pyspark/sql/streaming.py]: LGTM otherwise

**Comment by tdas** [comment]
Merged it in the hope of making it to the RC2. Will address comments in a different PR. Thanks @jose-torres and @HyukjinKwon for reviewing.

**Pull request closed without merging**